### ISSUE TITLE
Bug - Select: options lose focus styles when using children prop on typeahead

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#7802]
**Describe the problem**
When using the `children` prop for the SelectOption component to provide an alternate display, keyboard focus does not work correctly for the typeahead variant.

**How do you reproduce the problem?**
1. Go to [Select typeahead variant](https://www.patternfly.org/v4/components/select#typeahead)
2. Replace lines 131-137 with the below code
3. Try using keyboard to navigate the options when the menu is open, and notice that the options no longer have focus styles

```
{options.map((option, index) => (
  <SelectOption
    isDisabled={option.disabled}
    key={index}
    value={option.value}
    {...(option.description && { description: option.description })}
  >Option: {option.value}</SelectOption>
))}
```

**Expected behavior**
Focus style remain when using the `children` prop for an alternate display with typeahead variant

**Is this issue blocking you?**
N/A

**Screenshots**
N/A

**What is your environment?**
N/A

**What is your product and what release date are you targeting?**
N/A

**Any other information?**
N/A

**Comment by thatblindgeye** [comment]
Closing as the Next iteration of the Select component will not have the same logic for this issue to exist.
